src/server/api/routers/map_mod_publisher/mod.ts

```diff
@@ -477,6 +507,52 @@ const getGamebananaModInfo = async (gamebananaModID: number): Promise<Gamebanana
 
 
 
+const getModSearchDatabaseInfoForMod = async (gamebananaModID: number): Promise<ModSearchDatabase_ModInfo_NoId_Partial> => {
+    const modSearchDatabase = await getCurrentModSearchDatabase();
+
+
+    let modInfo: ModSearchDatabase_ModInfo_NoId_Partial | undefined = undefined;
+
+    for (const mod of modSearchDatabase) {
+        if (mod.GameBananaId === gamebananaModID) {
+            modInfo = {
+                Screenshots: mod.Screenshots,
+                Files: mod.Files,
+            };
+
+            break;
+        }
+    }
+
+
+    return modInfo ?? {};
+};
+
+
+const getModSearchDatabaseInfoForAllMods = async (mods: TrimmedMod[]): Promise<TrimmedMod[]> => {
+    const modSearchDatabase = await getCurrentModSearchDatabase();
+
+
+    outerLoop: for (const mod of mods) {
```

Comment:
This might be a slow loop if the mods list is large. It might be better to create a map from id to mod for modSearchDatabase and then use that for mods.

Reply:
Good thinking. Will fix.